src/main/scala/chisel3/annotations/annotations.scala

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.annotations
+
+import firrtl.annotations._
+import firrtl.FirrtlUserException
+
+/** The lowering convention of a module
+  * Controls how a module's ports are lowered to verilog.
+  */
+sealed class Convention(val string: String) {
+  def serialize: String = string
+}
+
+
+object Convention {
+  /** Undefined internal convention */
+  case object Internal extends Convention("internal")
+  /** Fully scalarize aggregate ports */
+  case object Scalarized extends Convention("scalarized")
+
+  def deserialize(string: String): Convention = string match {
+    case "internal" => Internal
+    case "scalarized" => Scalarized
+    case _ => throw new FirrtlUserException(s"Unrecognized Convention: $string")
+  }
+}
+
+/** Annotation to force the compiler to use a specified port convention.
+  * May only be applied to modules.
+  */
+case class ConventionAnnotation(target: ModuleName, convention: String)
```

Comment:
Can't store the convention as the enum version, has to be a string, because we cannot extend the anno serialization code, which is buried in firrtl.